src/Client.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Http\Client;
+
+/**
+ * Interface for both HTTP Client type
+ *
+ * @author Márk Sági-Kazár mark.sagikazar@gmail.com>
+ */
+interface Client extends PsrHttpClient, HttpClient
```

Comment:
Partly agree: we have something called ClientTemplate which implements all HTTP method using send. However I am not sure about the other. When the two interfaces are implemented together, `send` should use a message factory to create a request object then use `sendRequest` to actually send the request.